data/campaigns/tutorial/scenarios/02_Tutorial_part_2.cfg

```diff
@@ -325,6 +325,74 @@ A full list of abilities and weapons specials, along with traits, may be found i
         [/message]
     [/event]
 
+    # Triggers the first time that the player has two healers
+    [event]
+        name=recruit
+        [filter]
+            side=1
+            type=Elvish Shaman
+        [/filter]
+        [filter_condition]
+            [have_unit]
+                ability_type=heals
+                count=2
+            [/have_unit]
+        [/filter_condition]
+
+        [store_unit]
+            [filter]
+                ability_type=heals
+                [not]
+                    id=$unit.id
+                [/not]
+                {EVERYWHERE}
+            [/filter]
+            variable=other_healer
+        [/store_unit]
+
+        [if]
+            [variable]
+                name=other_healer.type
+                equals=Elvish Shaman
+            [/variable]
+            [then]
+                [message]
+                    speaker=$unit.id
+                    # po: both speaker and $other_healer are elvish shamans.
+                    # If there are allied sides then this isn't accurate, but at this stage of the tutorial allied sides haven't been introduced yet.
+                    message= _ "Both Shaman $other_healer.name and I can heal other units. However, each unit can only receive healing from one healer — even if we’re both next to a unit then that unit will still only heal 4 hitpoints per turn."
+                [/message]
+                [message]
+                    speaker=$unit.id
+                    # This line could be said by the other healer, but there's only one shaman portrait available, and it would be confusing to have dialog between two units with identical portraits.
+                    message= _ "If either of us advances to become a Druid, then a unit next to both of us will receive the Druid’s advanced healing. A unit in a village will receive healing from the village, neither a Shaman nor a Druid can improve on that."
```

Comment:
The second thing that this line says - that villages, shamans, and druids don't stack - how about saying it as soon as the _first_ shaman is recruited? Village healing had already been introduced by that point. That'll simplify _this_ dialog. (By that point rest healing has also been introduced, and it _does_ stack with other forms of healing, but I'm not sure it's worth pointing that out. Too much detail and the player won't see the forest for the trees - no pun intended.)

The fact that Druids have advanced healing is mentioned in a drive by, as though it's just reminding the player of something they already know. Maybe move it to front and center? It's also repeated in Galdrad's line just below. Maybe combine the two mentions of "Druids have advanced healing", the one in this line and the one in the next line? I don't have a concrete proposal but I can try to come up with on if you like.

Reply:
When the first shaman is recruited, there's already 3 `[message]`s, I don't think any more information can fit in that event.

I agree completely with your second paragraph.

Reply:
> When the first shaman is recruited, there's already 3 `[message]`s, I don't think any more information can fit in that event.

Agreed. And it doesn't seem to fit in S1, either:

https://github.com/wesnoth/wesnoth/blob/9a4bd1342a41a24f7fdae6ba6babc5dee72d8486/data/campaigns/tutorial/scenarios/01_Tutorial_part_1.cfg#L647
https://github.com/wesnoth/wesnoth/blob/9a4bd1342a41a24f7fdae6ba6babc5dee72d8486/data/campaigns/tutorial/scenarios/01_Tutorial_part_1.cfg#L844

Still, trying to cram too much information into a single line here won't do, either. I think it'll be best to first write down what all the points we want to make are ("multiple healers/villages don't stack", "multiple healers can be used in separate areas of the battle", etc) and then see how many of them we can make. We may have to leave one out of this conversation and find a place for it later in the scenario.

Reply:
We can speak about healers/villages/stacking here:

https://github.com/wesnoth/wesnoth/blob/a8c96ba1f45f0659de0db8dd2bb4c184f0ccbd84/data/campaigns/tutorial/scenarios/02_Tutorial_part_2.cfg#L705-L713

Reply:
Suggestion:

> Remember to retreat your wounded units to villages. Villages heal units faster than Elvish Shamans do: a unit in a village will be healed 8 hitpoints per turn, whereas a unit next to a Shaman will only be healed 4 points per turn.
> 
> Different sources of healing are not combined. If a unit starts its turn next to multiple healers, or on a village and next to a healer, it will still be healed only once.
> 
> There are two exceptions to this rule. First, if a unit doesn't do anything for a turn, it will heal 2 hitpoints <i>in addition</i> to any healing it may receive from villages or healers, for a total of up to 10 hitpoints per turn. The other exception involves allies — a topic which we will introduce later.

Reply:
> I think it'll be best to first write down what all the points we want to make are
> * "multiple healers/villages don't stack"
> * "multiple healers can be used in separate areas of the battle"

* druids heal more than shamans
* although they don't stack, the best one is used
* healers don't regen, but a pair of healers can heal each other
* rest healing does stack
* if you have a choice of which one to move, it's better to move the healer to the injured unit so that the injured unit rest-heals
* ... but if the unit is on the front line it probably needs to retreat

Reply:
- If you have two different healers, you can move one healer from a unit and put another one next to the unit, and it'll still enjoy rest healing
- Healers heal all units next to them, not just one
- Healers are important to keep alive, especially those that can advance to stronger healers

I don't think we'll be able to work all these details into the scenario. We should introduce the mechanic and outline how it's usually used, but we may have to defer some of the details or specific circumstances to tutorial S3 (#4399).

On the other hand, we have several places to fit dialog in:

- `turn 6` dialog about retreating
- recruiting the first Shaman
- recruiting the second Shaman
- first time Shaman/village healing happens

We can continue chatting on IRC if you prefer.

Reply:
Just realised that healing is well documented in the help's Gameplay section, so we can say where to look for more detail.

Reply:
Looking at the second `[message]` shown when recruiting the first Shaman, I think only one of these four sentences is on-topic:
https://github.com/wesnoth/wesnoth/blob/0b800eb8c5c3548330fed41ccdba8baad11cbcfa/data/campaigns/tutorial/scenarios/02_Tutorial_part_2.cfg#L315
I'm thinking of replacing that with this, and the warning about being careful would fit better in the first `[message]`'s  comment about being weak.
```
            caption= _ "The <i>Heal</i> Ability"
            message= _ "At the start of each turn, allied units adjacent to a Shaman will recover 4 hitpoints. This is only half of the amount that a village would provide, but you rarely have a village near your front-line. The Shaman doesn’t gain experience from healing, only from combat, but advancing her is worthwhile — her second-level Elvish Druid advancement has healing equivalent to a village.

A detailed explanation of healing may be found in the help (default hotkey: <b>F1</b>), in the <i>Gameplay</i> section."
```